plugins/storage/volume/linstor/src/main/java/org/apache/cloudstack/storage/datastore/driver/LinstorPrimaryDataStoreDriverImpl.java

```diff
@@ -965,6 +981,36 @@ private Answer copyTemplate(DataObject srcData, DataObject dstData) {
         return answer;
     }
 
+    private Answer copyVolume(DataObject srcData, DataObject dstData) {
+        VolumeInfo srcVolInfo = (VolumeInfo) srcData;
+        final StoragePoolVO pool = _storagePoolDao.findById(srcVolInfo.getDataStore().getId());
+        final DevelopersApi api = LinstorUtil.getLinstorAPI(pool.getHostAddress());
+        final String rscName = LinstorUtil.RSC_PREFIX + srcVolInfo.getUuid();
+
+        int nMaxExecutionMinutes = NumbersUtil.parseInt(
+                _configDao.getValue(Config.SecStorageCmdExecutionTimeMax.key()), 30);
```

Comment:
does this copy performed by ssvm? check if this is suitable config - `copy.volume.wait`

Reply:
TBH, I don't know exactly where the actual copy ends up.
I think it is used in the KVMStorageProcessor and will just use qemu-img convert to move it into secondary storage from Linstor.
So is here the `SecStorageCmdExecutionTimeMax` or `copy.volume.wait` more fitting?

Reply:
default for `SecStorageCmdExecutionTimeMax` is 30 minutes.
default for `CopyVolumeWait` is 10800 secs (180 minutes)

by the name of your method `CopyVolumeWait` seems more appropriate, but you could even create your own var (or use either)

Reply:
> `copy.volume.wait`

`copy.volume.wait` seems appropriate

Reply:
done